### ISSUE TITLE
Minikube fails to build ISO in Docker: cannot find package "k8s.io/minikube/pkg/minikube/cluster"

### ISSUE THREAD

**Issue by alban** [kubernetes/minikube#11902]
**Steps to reproduce the issue:** 

Following https://minikube.sigs.k8s.io/docs/contrib/building/iso/:

1. `git checkout v1.22.0-beta.0` (same problem on master as of 27d738657c9c7b1ae4578b28c094a57c8ad1f7f2)
2. `make buildroot-image`
3. `make out/minikube.iso`

**Full output of the build logs:**
<details>

```
[alban@neptune minikube]$ make out/minikube.iso
docker run --rm --workdir /mnt --volume /home/alban/go/src/k8s.io/minikube:/mnt  \
	--user 1000:1000 --env HOME=/tmp --env IN_DOCKER=1 \
	gcr.io/k8s-minikube/buildroot-image /usr/bin/make out/minikube.iso
/usr/bin/make minikube_iso
make[1]: Entering directory '/mnt'
GOOS=linux GOARCH=amd64 go build -o deploy/iso/minikube-iso/board/coreos/minikube/rootfs-overlay/usr/bin/auto-pause cmd/auto-pause/auto-pause.go
cmd/auto-pause/auto-pause.go:26:2: cannot find package "k8s.io/minikube/pkg/minikube/cluster" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/cluster (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/cluster (from $GOPATH)
cmd/auto-pause/auto-pause.go:27:2: cannot find package "k8s.io/minikube/pkg/minikube/command" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/command (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/command (from $GOPATH)
cmd/auto-pause/auto-pause.go:28:2: cannot find package "k8s.io/minikube/pkg/minikube/cruntime" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/cruntime (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/cruntime (from $GOPATH)
cmd/auto-pause/auto-pause.go:29:2: cannot find package "k8s.io/minikube/pkg/minikube/exit" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/exit (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/exit (from $GOPATH)
cmd/auto-pause/auto-pause.go:30:2: cannot find package "k8s.io/minikube/pkg/minikube/out" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/out (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/out (from $GOPATH)
cmd/auto-pause/auto-pause.go:31:2: cannot find package "k8s.io/minikube/pkg/minikube/reason" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/reason (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/reason (from $GOPATH)
cmd/auto-pause/auto-pause.go:32:2: cannot find package "k8s.io/minikube/pkg/minikube/style" in any of:
	/usr/lib/go-1.10/src/k8s.io/minikube/pkg/minikube/style (from $GOROOT)
	/tmp/go/src/k8s.io/minikube/pkg/minikube/style (from $GOPATH)
make[1]: *** [deploy/iso/minikube-iso/board/coreos/minikube/rootfs-overlay/usr/bin/auto-pause] Error 1
Makefile:875: recipe for target 'deploy/iso/minikube-iso/board/coreos/minikube/rootfs-overlay/usr/bin/auto-pause' failed
make[1]: Leaving directory '/mnt'
make: *** [out/minikube.iso] Error 2
Makefile:302: recipe for target 'out/minikube.iso' failed
make: *** [Makefile:304: out/minikube.iso] Error 2
```

</details>

**Additional logs**

<details>

```
[alban@neptune minikube]$ make iso_in_docker
docker run -it --rm --workdir /mnt --volume /home/alban/go/src/k8s.io/minikube:/mnt  \
	--user 1000:1000 --env HOME=/tmp --env IN_DOCKER=1 \
	gcr.io/k8s-minikube/buildroot-image /bin/bash
groups: cannot find name for group ID 1000
I have no name!@a466b28d3ef1:/mnt$ env|grep GO
I have no name!@a466b28d3ef1:/mnt$ go version
go version go1.10.4 linux/amd64
```
</details>

**Comment by medyagh** [comment]
@alban curious why are trying to build ISO ?

**Comment by alban** [comment]
@medyagh  I was trying to build the Minikube iso with a kernel that has the following options:
- CONFIG_IKHEADERS
- CONFIG_FTRACE_SYSCALLS
- CONFIG_DEBUG_INFO_BTF

See https://github.com/kubernetes/minikube/issues/8556

At the moment, we are using this fork to be able to make BPF workshops on Minikube:
https://github.com/kinvolk/cloud-native-bpf-workshop/blob/master/minikube.md

Thanks for the alternative make commands. I'll try `make release-iso`.

About the command `make iso_in_docker`, it was not working (see "Additional logs" section in the issue description)

**Comment by johnmuth81** [comment]
> @alban curious why are trying to build ISO ? in our automation we don't use building iso in docker

@medyagh  I'm not sure why @alban was trying to build the ISO, but I fell into `make out/minikube.iso` from https://minikube.sigs.k8s.io/docs/contrib/building/iso/ (which you can get to from the 'Contributing' link on the top README.md. I got further and found another issue with what appears to be a bogus SHA256 sum for CRIO 1.22.0, but I think I'll try `make iso-in-docker` instead.

**Comment by cameronbrunner** [comment]
I was trying the same thing and determined the issue to be 1.10 version of golang in the buildroot-image ISO Dockerfile.  Following the instructions at https://github.com/golang/go/wiki/Ubuntu I installed golang 1.17 to get past the initial error.  This led to a build error due to `PWD` not being set inside of the docker container due to the launching of `make` directly and not from inside a shell.  With a change to wrap the make call inside of a bash login prompt I was able to build an ISO on MacOS.

I will create a PR with changes.
